src/serac/numerics/functional/functional_qoi.inl

```diff
@@ -30,7 +30,10 @@ struct QoIProlongation : public mfem::Operator {
   /// @brief set the value of output to the distributed sum over input values from different processors
   void MultTranspose(const mfem::Vector& input, mfem::Vector& output) const override
   {
-    MPI_Allreduce(&input[0], &output[0], 1, MPI_DOUBLE, MPI_SUM, comm);
+    // save this to local variable to work around clang@14.0.6 compiler error:
```

Comment:
Not super excited about this workaround. We should see if there is a better way to handle this post PR.

Reply:
Is this solvable by a `const_cast`? https://godbolt.org/z/z3fKb8Pfb